### ISSUE TITLE
Latest commit causes an error message

### ISSUE THREAD

**Issue by blochl** [bronson/vim-trailing-whitespace#26]
The latest commit (9b472b1f6e825990ad8fa444142e749b55fe05aa) causes the following error message:

```
Error detected while processing /home/lbloch/.vim/plugged/vim-trailing-whitespace/plugin/trailing-whitespace.vim:
line   19:
E216: No such group or event: TermOpen * if ShouldMatchWhitespace() | match ExtraWhitespace /\\\@<![\u3000[:space:]]\+$/ | else | match ExtraWhitespace /^^/ | endif
```

System: fully updated Ubuntu 22.04.

**Comment by PabloScolpino** [comment]
Created https://github.com/bronson/vim-trailing-whitespace/pull/27 to fix this.

**Comment by PabloScolpino** [comment]
I'll try to fix this for both.

**Comment by PabloScolpino** [comment]
This solves the issue for both, but it is not clean.
https://github.com/bronson/vim-trailing-whitespace/pull/28

I am not a _vimscript_ expert by any means, so I'd like to see how this should be improved.

**Comment by superjamie** [comment]
I introduced this error, apologies.

airline found the same thing but they just removed Neovim terminal support altogether. That's a pretty undesirable solution: https://github.com/vim-airline/vim-airline/issues/1828

I tested Pablo's https://github.com/bronson/vim-trailing-whitespace/pull/28 on Neovim 0.8.3 and Vim 8.2.4919, it worked in both cases.

Another option is to string subsitute into an `exe` of the `autocmd` like this:

```vim
let term_open_event = (has('nvim') ? 'TermOpen' : 'TerminalOpen')
exe 'autocmd BufRead,BufNew,FileType,' term_open_event '* if ShouldMatchWhitespace() | match ExtraWhitespace /\\\@<![\u3000[:space:]]\+$/ | else | match ExtraWhitespace /^^/ | endif'
```

Maintainer's choice which syntax he prefers.
